types/seat/wlr_seat_pointer.c

```diff
@@ -274,6 +274,18 @@ void wlr_seat_pointer_send_axis(struct wlr_seat *wlr_seat, uint32_t time,
 		return;
 	}
 
+	bool send_source = false;
+	if (wlr_seat->pointer_state.sent_axis_source) {
+		if (wlr_seat->pointer_state.cached_axis_source != source) {
+			wlr_log(WLR_DEBUG,
```

Comment:
Maybe we should bump this to WLR_ERROR?

Reply:
Wouldn't that flood logs too much, writing an error message every pointer frame?

Reply:
Yeah, but this indicates a compositor bug. I'm afraid of not noticing it when it happens.

Reply:
Hmm, I guess a similar check should be added to Sway then.

Reply:
Hm, should it? The events come from the backends, I don't think there's a situation in which Sway arbitrarily changes the axis source or injects its own?

Reply:
You're right, there's no such situation, Sway just calls `wlr_seat_pointer_notify_axis()` which in turn calls `wlr_seat_pointer_send_axis()`. But it's a backend bug then, not compositor's.

Reply:
Maybe it should be changed to `assert()`?

Reply:
Yeah, I'm wondering about this as well. In general I'm in favor of failing hard when we hit inconsistent state, because that is guaranteed to result in a bug report. We can always downgrade to a log message if time tells this was a bad idea.